src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_evp.h

```diff
@@ -0,0 +1,23 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#pragma once
+
+#include "pal_jni.h"
+
+#define EVP_MAX_MD_SIZE 64
+
+PALEXPORT int32_t CryptoNative_EvpMdSize(intptr_t md);
+PALEXPORT int32_t CryptoNative_GetMaxMdSize(void);
+PALEXPORT intptr_t CryptoNative_EvpMd5(void);
+PALEXPORT intptr_t CryptoNative_EvpSha1(void);
+PALEXPORT intptr_t CryptoNative_EvpSha256(void);
+PALEXPORT intptr_t CryptoNative_EvpSha384(void);
+PALEXPORT intptr_t CryptoNative_EvpSha512(void);
+PALEXPORT int32_t CryptoNative_EvpDigestOneShot(intptr_t type, void* source, int32_t sourceSize, uint8_t* md, uint32_t* mdSize);
+PALEXPORT jobject CryptoNative_EvpMdCtxCreate(intptr_t type);
+PALEXPORT int32_t CryptoNative_EvpDigestReset(jobject ctx, intptr_t type);
+PALEXPORT int32_t CryptoNative_EvpDigestUpdate(jobject ctx, void* d, int32_t cnt);
+PALEXPORT int32_t CryptoNative_EvpDigestFinalEx(jobject ctx, uint8_t* md, uint32_t* s);
+PALEXPORT int32_t CryptoNative_EvpDigestCurrent(jobject ctx, uint8_t* md, uint32_t* s);
+PALEXPORT void CryptoNative_EvpMdCtxDestroy(jobject ctx);
```

Comment:
Unless there's a goal that the Android shim be able to be drop-in substituted for the OpenSSL shim (which feels like a hard state to maintain), the functions in this library should be named more appropriate to their context.

* CryptoNative is OK as long as this library and the OpenSSL shim can't ever be loaded in the same process, otherwise the exported function prefix for this library should be different (e.g. AndroidCryptoNative).
* Names like EvpDigestUpdate are because it's a thin wrapper over EVP_DigestUpdate.  Since the Android implementation doesn't involve a function of that name the name should be more reflective of what is going on.

So I'd expect CryptoNative_EvpDigestUpdate to be AndroidCryptoNative_MessageDigestUpdate

Reply:
@bartonjs thats the plan, but I'd like to do it a bit later when it will be clear that we can replace openssl.